### PR TITLE
Add support for named lambda request handlers

### LINKED ISSUE #3099: Multiple handlers for amazon-lambda extension
**Description**
Currently, only a single Handler/Function is allowed in the Amazon Lambda extension. Meaning you would have to build a single artifact/zipfile for each lambda. If you have a large project consisting of dozens of endpoints/lambdas, building all this would take forever. It should be possible to build one artifact with multiple lambdas and then specify which handler you want to to use when deploying.

**Implementation ideas**
See the following: https://github.com/marcinczeczko/quarkus-lambda-native/issues/1#issuecomment-508665645

### PR DESCRIPTION
**Pull request by stuartwdouglas**
Fixes #3099

### CODE REVIEW

File: extensions/amazon-lambda/deployment/src/main/java/io/quarkus/amazon/lambda/deployment/AmazonLambdaProcessor.java [hunk 1]
```diff
@@ -33,16 +39,28 @@
...
+            builder.addBeanClass(name.toString());
+            String cdiName = null;
+            List<AnnotationInstance> named = info.annotations().get(NAMED);
+            if (named != null && !named.isEmpty()) {
+                cdiName = named.get(0).value().asString();
+            }
 
             final String lambda = name.toString();
-            ret.add(new AmazonLambdaClassNameBuildItem(lambda));
+            ret.add(new AmazonLambdaBuildItem(lambda, cdiName));
```
Comment (evanchooly) [inline extensions/amazon-lambda/deployment/src/main/java/io/quarkus/amazon/lambda/deployment/AmazonLambdaProcessor.java]: what does naming the handler give us exactly?  this feels like a new requirement.

File: extensions/amazon-lambda/deployment/src/main/java/io/quarkus/amazon/lambda/deployment/AmazonLambdaProcessor.java [hunk 2]
```diff
@@ -106,22 +102,23 @@ void bootstrap(BuildProducer<GeneratedResourceBuildItem> generatedResources) thr
 
     @BuildStep
     @Record(ExecutionTime.RUNTIME_INIT)
-    public void servlets(List<AmazonLambdaBuildItem> lambdas,
+    public void processLambas(List<AmazonLambdaBuildItem> lambdas,
```
Comment (evanchooly) [inline extensions/amazon-lambda/deployment/src/main/java/io/quarkus/amazon/lambda/deployment/AmazonLambdaProcessor.java]: `processLambdas`

File: extensions/amazon-lambda/deployment/src/main/java/io/quarkus/amazon/lambda/deployment/AmazonLambdaProcessor.java (hunk 1 above)
Comment (evanchooly) [reply extensions/amazon-lambda/deployment/src/main/java/io/quarkus/amazon/lambda/deployment/AmazonLambdaProcessor.java]: i think i figured it out below.  :)

**Review by evanchooly — APPROVED** [review]
This makes sense so far.  Looking at the tests in `integration-tests`, I see they're still "powered" by undertow.  should all that be updated to use the vert.x/netty bits?

**Comment by stuartwdouglas** [comment]
On Fri, 4 Oct 2019 at 05:48, Justin Lee <notifications@github.com> wrote:

> *@evanchooly* approved this pull request.
>
> This makes sense so far. Looking at the tests in integration-tests, I see
> they're still "powered" by undertow. should all that be updated to use the
> vert.x/netty bits?\
>

No, this is just a test framework that emulates the AWS environment. It is
an internal thing that is not actually using Quarkus.

Stuart

**Pull request merged**